_posts/2019-06-07-python-datascience-with-openfaas.md

```diff
@@ -0,0 +1,267 @@
+2. It also means that each new deployment is completely versioned, you know the version of the model from the Docker tag. This makes rolling backward or forward as simple as changing the Docker tag on the deployment.
+3. One final benefit is that startup time can be slightly faster because once a node has the Docker image in its cache, it has everything it needs to start new instances of the function.  If you load the model from an external source, then every function instance that starts _must_ curl/copy from that external location, there is no way to cache it for new instances.
+
+On the other hand, this does result in slightly larger images, in this case the model is 107k. This size will, of course, depend on the libraries being used and the type of model being serialized.  For many models, the simplicity of a self-contained image out-weighs the cost of the additional size. Often Docker layer caching will hide almost all of it.
+
+You can try the latest version of this function in your own cluster using
+
+```bash
+faas-cli deploy --image=theaxer/classify:latest --name=classify
+```
+
+## A note about Python 2 deprecation
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Remember that Python 2 will reach end of life on 1/1/2020. For the impact on *your* project, see <a href="https://t.co/giiS9CNn8V">https://t.co/giiS9CNn8V</a></p>&mdash; Guido van Rossum (@gvanrossum) <a href="https://twitter.com/gvanrossum/status/1133496146700058626?ref_src=twsrc%5Etfw">May 28, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+You will not find a Python 2 template in the [pydatascience-template][pydata-template] repo because we are coming very quickly to the end-of-life for Python 2.  If you are still using Python 2, the templates could be forked and modified, but please be careful. Most packages will start deprecating Python 2 support soon.
+
+
+## Wrapping up
+
+Data science is of course a popular topic these days and OpenFaaS can help [simplify, standardize, and speed up][kubecon-bt-journey] the deployment process for your team. This post introduces just one example of what a data science focused template could look like. If you are building and deploying data science models, let us know what you think through the [Slack community][of-slack] or [Twitter][of-twitter].
+
+
+### Going further
+Want to help improve the template, join our slack and ping `@Lucas Roesler` or open an [issue on Github][name-classifier-issues].
+
+PyTorch has also provided a [C++ frontend][pytorch-cpp] that allows deploying the serialized model as a C++ function.  There is also a [Go frontend][pytorch-golang] in progress.  The `pydatascience-template` is designed to be reusable for any data science library, but it would be interesting to see what kind of optimizations could be created using one of these other frontends for deployment.
```

Comment:
`PyTorch has also provided a [C++ frontend][pytorch-cpp] that allows deploying the serialized model as a C++ function. `

Prefer plain English:

`You can use PyTorch's C++ frontend  to deploy your serialised model as a C++ function.`